server/boot/validations.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+
+var idEquals = require("loopback-datasource-juggler/lib/utils").idEquals;
+
+exports.efficientUniqueness = efficientUniqueness;
+exports.idsHaveDuplicates = idsHaveDuplicates;
+exports.sanitazeUniqueness = sanitazeUniqueness;
+
+/**
+ * Check a list of IDs to see if there are any duplicates.
+ *
+ * @param {Array} The array of IDs to check
+ * @returns {boolean} If any duplicates were found
+ */
+function idsHaveDuplicates(ids) {
+  // use Set if available and all ids are of string or number type
+  var hasDuplicates = undefined;
+  var i, j;
+  if (typeof Set === "function") {
+    var uniqueIds = new Set();
+    for (i = 0; i < ids.length; ++i) {
+      var idType = typeof ids[i];
+      if (idType === "string" || idType === "number") {
+        if (uniqueIds.has(ids[i])) {
```

Comment:
I'm not sure when an id would come in as a number or a string. Would you want to convert to string for the check If it's possible that the same id could come in  as "42" or 42 but represent the same id? Or in that case would it be inherently a different id?

Reply:
I must admit this piece of code comes from the link I posted, https://github.com/loopbackio/loopback-datasource-juggler/pull/1629, so it's basically what happens in loopback4. For that reason, I wouldn't change it, but I'm open to suggestions!

Reply:
@dylanmcreynolds : I would agree with @minottic . Let's include this code as it is for now. We can always change it later.